### PR TITLE
Fix some issues which were reported by Codacy

### PR DESCRIPTION
**Pull request by stweil**

### CODE REVIEW

**Comment by stweil** [comment]
According to Codacy this pull request fixes about 32 issues (from a total of currently 590 issues), so much more work remains to be done.

**Comment by egorpugin** [comment]
Is it possible to use auto in more places?

File: src/ccmain/docqual.cpp [hunk 1]
```diff
@@ -284,9 +280,9 @@ void Tesseract::doc_and_block_rejection( // reject big chunks
         }
 
         /* Walk rows in block testing for row rejection */
-        row_no = 0;
+        int16_t row_no = 0;
         while (page_res_it.word() != nullptr && page_res_it.block() == current_block) {
-          current_row = page_res_it.row();
+          ROW_RES *current_row = page_res_it.row();
```
Comment (egorpugin) [inline src/ccmain/docqual.cpp:285]: auto?

File: src/ccmain/docqual.cpp [hunk 2]
```diff
@@ -539,7 +533,7 @@ void Tesseract::tilde_delete(PAGE_RES_IT &page_res_it) {
 
   page_res_it.restart_page();
   while (page_res_it.word() != nullptr) {
-    word = page_res_it.word();
+    WERD_RES *word = page_res_it.word();
```
Comment (egorpugin) [inline src/ccmain/docqual.cpp:536]: auto?

File: src/ccmain/fixspace.cpp [hunk 3]
```diff
@@ -183,7 +182,7 @@ void Tesseract::fix_fuzzy_space_list(WERD_RES_LIST &best_perm, ROW *row, BLOCK *
 
   while ((best_score != PERFECT_WERDS) && !current_perm.empty()) {
     match_current_words(current_perm, row, block);
-    current_score = eval_word_spacing(current_perm);
+    int16_t current_score = eval_word_spacing(current_perm);
```
Comment (egorpugin) [inline src/ccmain/fixspace.cpp:185]: auto?

File: src/ccmain/fixspace.cpp [hunk 4]
```diff
@@ -201,11 +200,10 @@ void Tesseract::fix_fuzzy_space_list(WERD_RES_LIST &best_perm, ROW *row, BLOCK *
 void initialise_search(WERD_RES_LIST &src_list, WERD_RES_LIST &new_list) {
   WERD_RES_IT src_it(&src_list);
   WERD_RES_IT new_it(&new_list);
-  WERD_RES *src_wd;
   WERD_RES *new_wd;
 
   for (src_it.mark_cycle_pt(); !src_it.cycled_list(); src_it.forward()) {
-    src_wd = src_it.data();
+    WERD_RES *src_wd = src_it.data();
```
Comment (egorpugin) [inline src/ccmain/fixspace.cpp:206]: auto?

**Review by egorpugin — APPROVED** [review]

**Comment by stweil** [comment]
> Is it possible to use auto in more places?

Yes, sure, but sometimes I want to be able to see the data types explicitly for myself, and in other cases more work is needed to verify that `auto` or `auto *` really gives the same result.

**Pull request merged**